epilogue-processor/src/main/java/edu/wpi/first/epilogue/processor/LoggerGenerator.java

```diff
@@ -346,12 +354,12 @@ private void writeLoggerFile(
                     var logInvocation = h.logInvocation(loggableElement, clazz);
                     if (logInvocation != null) {
                       // Generate log invocation for the main element
-                      out.println(logInvocation.indent(6).stripTrailing() + ";");
-
-                      // Generate log invocations for dependencies if any
-                      var dependencyInvocations = generateDependencyLogging(loggableElement, clazz);
-                      for (var dependencyInvocation : dependencyInvocations) {
-                        out.println(dependencyInvocation.indent(6).stripTrailing() + ";");
+                      var metadata = generateLogMetadata(loggableElement, clazz);
+                      if (metadata != null) {
+                        var modifiedInvocation = logInvocation.replaceFirst("\\)$", ", " + metadata + ")");
```

Comment:
It'd be better to have the element handler generate the metadata arguments instead of manipulating the return values after the fact; it can be tricky to handle all possible cases with string manipulation.

Splitting the metadata to be declared as a field also means the element handler can just emit `element.getSimpleName() + "Metadata"` without needing to know about anything the metadata depends upon